src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -1727,12 +1791,444 @@ def get_no_wait(self, **kwargs) -> bool:
+        CONST_OUTBOUND_TYPE_LOAD_BALANCER.
+
+        This function supports the option of load_balancer_profile, if provided, when verifying loadbalancer-related
+        parameters, the value in load_balancer_profile will be used for validation.
+        This function supports the option of enable_validation. When enabled, if the value of outbound_type is
+        userDefinedRouting (CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING), the following checks will be performed. If
+        load_balancer_sku is set to basic, an InvalidArgumentValueError will be raised. If vnet_subnet_id is not
+        assigned, a RequiredArgumentMissingError will be raised. If any of load_balancer_managed_outbound_ip_count,
+        load_balancer_outbound_ips or load_balancer_outbound_ip_prefixes is assigned, a MutuallyExclusiveArgumentError
+        will be raised.
+        This function supports the option of read_only. When enabled, it will skip dynamic completion and validation.
+
+        :return: string or None
+        """
+        # read the original value passed by the command
+        raw_value = self.raw_param.get(
+            "outbound_type"
+        )
+        # try to read the property value corresponding to the parameter from the `mc` object
+        value_obtained_from_mc = None
+        if (
+            self.mc and
+            self.mc.network_profile
+        ):
+            value_obtained_from_mc = (
+                self.mc.network_profile.outbound_type
+            )
+
+        # set default value
+        read_from_mc = False
+        if value_obtained_from_mc is not None:
+            outbound_type = value_obtained_from_mc
+            read_from_mc = True
+        else:
+            outbound_type = raw_value
+
+        # skip dynamic completion & validation if option read_only is specified
+        if kwargs.get("read_only"):
+            return outbound_type
+
+        dynamic_completion = False
+        # check whether the parameter meet the conditions of dynamic completion
+        if outbound_type != CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING:
+            dynamic_completion = True
+        # disable dynamic completion if the value is read from `mc`
+        dynamic_completion = dynamic_completion and not read_from_mc
+        if dynamic_completion:
+            outbound_type = CONST_OUTBOUND_TYPE_LOAD_BALANCER
+
+        # validation
+        # Note: The parameters involved in the validation are not verified in their own getters.
+        if enable_validation:
+            if outbound_type == CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING:
+                # Should not enable read_only for get_load_balancer_sku, since its default value is None, and it has
+                # not been decorated into the mc object at this time, only the value after dynamic completion is
+                # meaningful here.
+                if self.get_load_balancer_sku() == "basic":
```

Comment:
ignore case?

Reply:
Function "get_load_balancer_sku" will always return lowercase string or None. We have got unit test cases to cover "get_load_balancer_sku".